src/lib/snark_worker/entry.ml

```diff
@@ -198,9 +125,11 @@ let main (module Rpcs_versioned : Intf.Rpcs_versioned_S) ~logger ~proof_level
             in
             log_and_retry "performing work" e (retry_pause 10.) go
         | Ok result ->
-            emit_proof_metrics result.metrics
-              (Selector.Result.Stable.Latest.transactions result)
-              logger ;
+            One_or_two.zip_exn work.instances result.metrics
+            |> One_or_two.iter ~f:(fun (single_spec, (elapsed, _tag)) ->
```

Comment:
These code are meant to be temporary, with the whole protocol update they would be removed later. So it's fine that they're a bit ugly for now.